From c047667650f95d4f3db94787e131fca17ebd0212 Mon Sep 17 00:00:00 2001
From: xtex <xtex@astrafall.org>
Date: Sun, 8 Mar 2026 09:43:35 +0800
Subject: [PATCH 129/133] std.zig.system.loongarch: remove LBT detection
X-Developer-Signature: v=1; a=openpgp-sha256; l=1018; i=xtex@astrafall.org;
 h=from:subject; bh=paEvOyWRscZV+GeA+rOg8fnQorv40kNLb3IV09nffUo=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78gQea7/5873K/vQrwR/nZK8YvvvN8Koqs8Zwd4nLk
 biTrvfOdkxkYRDjYrAUU2QpMmzwZtVJ5xddVi4LM4eVCWSItEgDAwMDAwsDX25iXqmRjpGeqbah
 nqGhjpGOEQMXpwBM9ZnHDP9jSlP/SG7eyXa1a8bPifsLslz2B3Fu9Wf/dkxmmn7xnzN3GP7HsHQ
 dkL7Q84Wh6dG0dx0x8hruvxfuSHw4nS3E9yW74TduAA==
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

Compiler backends should not emit any LBT instructions
except for inline assemblies.
The binary translation extensions are for special
applications such as LATX, and should be explicitly enabled
when needed.

Change-Id: Ifd351e5a35980a7f374ab06e18673b0c7aa2da58
Signed-off-by: xtex <xtex@astrafall.org>
---
 lib/std/zig/system/loongarch.zig | 2 --
 1 file changed, 2 deletions(-)

diff --git a/lib/std/zig/system/loongarch.zig b/lib/std/zig/system/loongarch.zig
index 30d48a725138..d9a61b10a7da 100644
--- a/lib/std/zig/system/loongarch.zig
+++ b/lib/std/zig/system/loongarch.zig
@@ -50,8 +50,6 @@ pub fn detectNativeCpuAndFeatures(
     setFeature(&cpu, .lasx, (hwcap_bits & HWCAP.LASX) != 0);
     setFeature(&cpu, .lvz, (hwcap_bits & HWCAP.LVZ) != 0);
 
-    setFeature(&cpu, .lbt, bit(cfg2, 18) and bit(cfg2, 19) and bit(cfg2, 20));
-
     setFeature(&cpu, .frecipe, bit(cfg2, 25));
     setFeature(&cpu, .div32, bit(cfg2, 26));
     setFeature(&cpu, .lam_bh, bit(cfg2, 27));
-- 
2.52.0

